### PR TITLE
Fix using_task_library dependency resolution

### PR DESCRIPTION
**Pull request by hemker**
When using a "separate prefixes" build configuration, included headers from a "using_task_library" dependency were not found. I am not 100% sure whether my fix is totally correct to fix the problem, but as far is I understood the `used_task_libraries` member of the project was never used to generate the task's dependencies in that specific case, so I added them.

### CODE REVIEW

**Comment by doudou** [comment]
Hi @hemker 

Could you maybe give us a specific error you are getting ? I am not sure what`s the actual issue.

Just to frame the discussion: what the current code is doing is only include task libraries that are actually used (the two lines before your change) - whether this is a good thing is arguable.

**Comment by hemker** [comment]
Hey @doudou
I just provided a test case to reproduce the problem. In the test module `dependent_includes` you can find the include `#include <cross_consumer/TaskStates.hpp>. The provided test case fails (because the header is not found) without having the extra line in project.rb.

File: test/data/modules/dependent_includes/tasks/DependentIncludes.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include "DependentIncludes.hpp"
+#include <cross_consumer/TaskStates.hpp>
```
Comment (hemker) [inline test/data/modules/dependent_includes/tasks/DependentIncludes.cpp:2]: Failing line without adaptions in project.rb

Comment (doudou) [reply test/data/modules/dependent_includes/tasks/DependentIncludes.cpp:2]: Thanks. The codegen test suite has always been crap. Nowadays we are using the rock build tests suite which is a lot more flexible.

For fail-to-build errors, just create the necessary orogen project(s) under `build_tests/orogen/` and add them to the buildconf.

**Comment by doudou** [comment]
So, I really can`t find a good reason why your change would be bad. And there's definitely a good reason for it...

Just implement the one small change in my review, and create the test package(s) separately. Name them over what they are testing, for instance build_tests/orogen/using_task_library-not_picking_up_not_directly_used_dependencies-publisher and build_tests/orogen/using_task_library-not_picking_up_not_directly_used_dependencies-consumer. And refer to the bug in the buildconf commit.

File: lib/orogen/gen/project.rb [hunk 2]
```diff
@@ -814,6 +814,7 @@ def tasklib_used_task_libraries
                     result = self_tasks.inject(Set.new) do |set, task|
                         set | task.used_task_libraries
                     end
+                    result.merge(used_task_libraries)
                     result.to_a.sort_by(&:name)
```
Comment (doudou) [inline lib/orogen/gen/project.rb:818]: `self#used_task_libraries` is a superset of the `result` created from `self_tasks`, so you can simply replace the whole method with `used_task_libraries.to_a.sort_by(&:name)`

**Comment by hemker** [comment]
@doudou Did all required steps.

File: lib/orogen/gen/project.rb [hunk 3]
```diff
@@ -811,12 +811,8 @@ def generate_build_system # :nodoc:
...
                 # depends on
                 def tasklib_used_task_libraries
-                    result = self_tasks.inject(Set.new) do |set, task|
-                        set | task.used_task_libraries
-                    end
-                    result.merge(used_task_libraries)
-                    result.to_a.sort_by(&:name)
-                end
+                    used_task_libraries.to_a.sort_by(&:name)
+		end
```
Comment (doudou) [inline lib/orogen/gen/project.rb:815]: Tabs vs spaces ... Please use 4 spaces, no tabs (Rock-wide convention)

**Comment by hemker** [comment]
@doudou Still anything to do here?

**Comment by doudou** [comment]
My apologies @hemker ... I'm on it now

**Pull request merged**